src/states/state-handlers.ts

```diff
@@ -228,6 +239,46 @@ export const handleLoggedIn = async (ctx: BotHandlerContext): Promise<User> => {
   return updatedUser;
 };
 
+export const handleReplyAttendanceButton = async (ctx: BotHandlerContext): Promise<User> => {
+  const { payload } = ctx;
+  const updatedUser = structuredClone(ctx.user);
+
+  // Check if the user clicked either "Today's Attendance" or "Yesterday's Attendance"
+  if (payload.button.text === "Today's Attendance" || payload.button.text === "Yesterday's Attendance") {
+    const today = new Date();
+    let selectedDate;
+
+    if (payload.button.text === "Today's Attendance") {
+      // User clicked "Today's Attendance"
+      selectedDate = today;
+    } else {
+      // User clicked "Yesterday's Attendance"
+      const yesterday = new Date(today);
+      yesterday.setDate(today.getDate() - 1);
+      selectedDate = yesterday;
+    }
+
+    try {
+      // Fetch attendance data for the selected date using your Amizone API client
+      const [year, month, day] = [selectedDate.getFullYear(), selectedDate.getMonth() + 1, selectedDate.getDate()];
+      const amizoneClient = newAmizoneClient(updatedUser.amizoneCredentials);
+      const attendance = await amizoneClient.amizoneServiceGetClassSchedule(year, month, day);
+
+      // Send the attendance data as a message to the user
+      await ctx.bot.sendMessage(payload.sender, renderSchedule(attendance.data));
+
+      // Update the user's state back to LOGGED_IN
+      updatedUser.state = states.LOGGED_IN;
```

Comment:
State should be `LOGGED_IN` already

Reply:
Thanks for the review! The code seems to work properly given the updations!